### PR TITLE
 🎨 Handles wallet forbidden error and enhances handling of unexpected errors

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

This PR is focused on enhancing error handling in the web API for unexpected errors and also specifically the managing wallet access errors during project patching.

### Key changes in the PR:

1. **Improved Wallet Access Error Handling**:
   - Issue: The PR addresses the problem reported in issue **#6443**, where wallet access errors, particularly when patching a shared project, were previously leading to a generic HTTP `500` response (Internal Server Error).
   - Improvement: The PR now properly handles these errors, ensuring that they are managed correctly, even though the underlying issue isn't fully resolved. Rather than simply returning a `500` error, the system will now handle and notify the user about the wallet access issue, providing better clarity on the failure point.

2. **`servicelib.aiohttp.rest_middleware` Updates**:
   - Error middleware in `servicelib.aiohttp.rest_middleware` has been updated to handle unhandled exceptions more gracefully.
   - When an unhandled exception occurs, the middleware now returns a `500 Internal Server Error` response, but logs the error using the function `troubleshooting_log_message`. This helps ensure that error details are logged effectively with associated error codes for easier diagnosis during troubleshooting.

3. ~**Front-End Handling of `500` Error Messages**~
   ~- The front-end is now equipped to properly handle `500` responses from the API. This means that when such errors occur, the front-end will display appropriate messages to the user, improving the overall user experience during failure scenarios.~
@odeimaiz For the moment I revert the front-end changes. We could follow up on that in place

### Implications and Benefits:
- **Clearer Error Reporting**: This PR improves the clarity of error responses, making it easier for both developers and users to understand the nature of the problem (in this case, wallet access errors).
- **Logging Improvements**: Using `troubleshooting_log_message` ensures that issues are logged more systematically, aiding in debugging and troubleshooting without leaving gaps.
- **Better User Experience**: With the front-end prepared to handle `500` errors, the user experience is less likely to be disrupted by vague or unclear error messages.

These improvements are a step forward in ensuring a more reliable and user-friendly API service, even in scenarios involving unexpected errors.

## Related issue/s

- handles error from #6443 
- Using tools from https://github.com/ITISFoundation/osparc-simcore/pull/6432

## Dev-ops checklist
None

### CODE REVIEW

File: services/static-webserver/client/source/class/osparc/FlashMessenger.js [hunk 1]
```diff
@@ -111,9 +111,16 @@ qx.Class.define("osparc.FlashMessenger", {
       }
       this.__displayedMessagesCount++;
 
+      const message = flashMessage.getMessage();
+      if (!message) {
+        // If message is "", null or undefined, do not show it
+        this.removeMessage(flashMessage);
+        return;
+      }
+
```
Comment (odeimaiz) [inline services/static-webserver/client/source/class/osparc/FlashMessenger.js:120]: Why don't you do this before adding it to the container?

File: services/static-webserver/client/source/class/osparc/data/Resources.js [hunk 2]
```diff
@@ -1301,23 +1301,13 @@ qx.Class.define("osparc.data.Resources", {
...
-            }
-            const errors = errorData.errors || [];
-            if (message === null && errors && errors.length) {
-              message = errors[0].message;
-            }
-            status = errorData.status;
-          } else {
-            const req = e.getRequest();
-            message = req.getResponse();
-            status = req.getStatus();
```
Comment (odeimaiz) [inline services/static-webserver/client/source/class/osparc/data/Resources.js:1320]: All this was here to handle all kind of different responses coming from the backend. Are you sure your cleanup is still able to handle them?

File: services/static-webserver/client/source/class/osparc/FlashMessenger.js (hunk 1 above)
Comment (pcrespov) [reply services/static-webserver/client/source/class/osparc/FlashMessenger.js:120]: reverting

File: services/static-webserver/client/source/class/osparc/data/Resources.js [hunk 3]
```diff
@@ -1335,12 +1325,19 @@ qx.Class.define("osparc.data.Resources", {
...
+          } else if ([404, 503].includes(status)) {
+            message += "<br>"+qx.locale.Manager.tr("Please try again later");
           }
-          const err = Error(message ? message : `Error while trying to fetch ${endpoint} ${resource}`);
-          if (status) {
-            err.status = status;
+          const err = Error(message);
+          err.status = status;
+          if (errorType) {
+            err.type = errorType
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/static-webserver/client/source/class/osparc/data/Resources.js:1340]: reverted

File: packages/service-library/src/servicelib/aiohttp/rest_middlewares.py [hunk 4]
```diff
@@ -28,6 +31,7 @@
 from .typing_extension import Handler, Middleware
 
 DEFAULT_API_VERSION = "v0"
+FMSG_INTERNAL_ERROR_USER_FRIENDLY = "Oops! Something went wrong, but we've noted it down and we'll sort it out ASAP. Thanks for your patience! [{}]"
```
Comment (GitHK) [inline packages/service-library/src/servicelib/aiohttp/rest_middlewares.py:34]: Let's try something slightly more formal and professional for our clients.

```suggestion
FMSG_INTERNAL_ERROR_USER_FRIENDLY = "We apologize for the inconvenience. Our team has recorded the issue and is working to resolve it as quickly as possible. Thank you for your patience! [{}]"
```

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks a lot for this!

File: packages/service-library/tests/aiohttp/test_rest_middlewares.py [hunk 5]
```diff
@@ -147,3 +212,62 @@ async def test_404_not_found(client: TestClient):
...
+    parsed_oec = parse_error_code(error["message"]).pop()
+    assert FMSG_INTERNAL_ERROR_USER_FRIENDLY.format(parsed_oec) == error["message"]
+
+    # avoids details
+    assert not error.get("errors")
+    assert not error.get("logs")
+
+    # - log sufficient information to diagnose the issue
+    #
+    # ERROR    servicelib.aiohttp.rest_middlewares:rest_middlewares.py:75 Oops! Something went wrong, but we've noted it down and we'll sort it out ASAP. Thanks for your patience! [OEC:128594540599840].
```
Comment (GitHK) [inline packages/service-library/tests/aiohttp/test_rest_middlewares.py:242]: please also update this comment wit the new formatted message

**Review by GitHK — APPROVED** [review]
👍 Please take into consideration my suggestion

**Pull request merged**